diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -411,32 +412,17 @@ controller_interface::CallbackReturn DiffDriveController::on_configure(
     std::make_shared<realtime_tools::RealtimePublisher<nav_msgs::msg::Odometry>>(
       odometry_publisher_);
 
-  // Append the tf prefix if there is one
-  std::string tf_prefix = "";
-  if (params_.tf_frame_prefix_enable)
-  {
-    if (params_.tf_frame_prefix != "")
-    {
-      tf_prefix = params_.tf_frame_prefix;
-    }
-    else
-    {
-      tf_prefix = std::string(get_node()->get_namespace());
-    }
-
-    // Make sure prefix does not start with '/' and always ends with '/'
-    if (tf_prefix.back() != '/')
-    {
-      tf_prefix = tf_prefix + "/";
-    }
-    if (tf_prefix.front() == '/')
-    {
-      tf_prefix.erase(0, 1);
-    }
-  }
-
-  const auto odom_frame_id = tf_prefix + params_.odom_frame_id;
-  const auto base_frame_id = tf_prefix + params_.base_frame_id;
+  // apply the TF prefix if not empty, otherwise use the node namespace
+  const auto odom_frame_id =
+    params_.tf_frame_prefix_enable
+      ? control_toolbox::apply_tf_prefix(
+          params_.tf_frame_prefix, get_node()->get_namespace(), params_.odom_frame_id)
+      : params_.odom_frame_id;
```

Comment:
should we change the API to only return the prefix, and just append the  odom_rame_id in any case?
something like this?
```suggestion
    (params_.tf_frame_prefix_enable
      ? control_toolbox::get_tf_prefix(
          params_.tf_frame_prefix, get_node()->get_namespace()) : "")
      + params_.odom_frame_id;
```

Reply:
Sure no problem